Simplified/Reader2/AdobeDRM/AdobeDRMContentProtection.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  AdobeDRMContentProtection.swift
+//  Simplified
+//
+//  Created by Vladimir Fedorov on 20.01.2021.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import R2Shared
+
+#if FEATURE_DRM_CONNECTOR
+
+class AdobeDRMContentProtection: ContentProtection {
+  func open(asset: PublicationAsset, fetcher: Fetcher, credentials: String?, allowUserInteraction: Bool, sender: Any?, completion: @escaping (CancellableResult<ProtectedAsset?, Publication.OpeningError>) -> Void) {
+    
+    // Publication asset must be a FileAsset, as we are opening a file
+    // If not, we can't open and use Adobe DRM
+    guard let fileAsset = asset as? FileAsset else {
+      completion(.failure(.unavailable(nil)))
+      return
+    }
+    
+    do {
+      // META-INF is a part of .epub structure
+      let encryptionPath = "META-INF/encryption.xml"
```

Comment:
do you have docs/specs that describe this encryption.xml path? It would be good to reference them here.

Reply:
it's in ADEPT code, there's no good documentation about this file, except it must be present in .epub for DRM to work. File data (it describes encryption algorithms used to encrypt each file) is passed to Adobe DRM decryption functions.